packages/atomic-hosted-page/src/components/atomic-hosted-ui/atomic-hosted-ui.tsx

```diff
@@ -0,0 +1,107 @@
+import {StringValue} from '@coveo/bueno';
+import {
+  PlatformEnvironment,
+  getOrganizationEndpoints as getOrganizationEndpointsHeadless,
+} from '@coveo/headless';
+import {
+  Component,
+  ComponentInterface,
+  Method,
+  Element,
+  Prop,
+} from '@stencil/core';
+import {
+  InitializationOptions,
+  extractPlatformUrl,
+  validateOptions,
+} from '../utils/options-utils';
+import {processHostedPage} from './hosted-ui';
+
+interface AtomicHostedUIInitializationOptions extends InitializationOptions {
+  /**
+   * The unique identifier of the hosted search page.
+   */
+  pageId: string;
+}
+
+/**
+ * A Web Component used to inject a Coveo Hosted Search Page in the DOM.
+ * Pulls from the [Search Interfaces API](https://platform.cloud.coveo.com/docs?urls.primaryName=Search%20Interface%20Service#/)
+ * @internal
+ */
+@Component({
+  tag: 'atomic-hosted-ui',
+  shadow: false,
+})
+export class AtomicHostedUI implements ComponentInterface {
+  @Element() private element!: HTMLElement;
+
+  private validateOptions(opts: AtomicHostedUIInitializationOptions) {
+    validateOptions(opts, {
+      pageId: new StringValue({required: true, emptyAllowed: false}),
+    });
+  }
+
+  /**
+   * The type of hosted search page to load
+   * E.g., 'page', 'custom', 'legacy'
```

Comment:
```suggestion
```

Since the `@Prop` decorator has `reflect: true`, I believe this information is redundant.

Reply:
I understand that with reflect, the attributes will be rendered in the DOM, but don't the user still need to know which options are valid?

Reply:
There's no harm in leaving them here, but I'm fairly certain the information will be duplicated in the generated reference documentation.

E.g., look at unit-display in the table here: https://docs.coveo.com/en/atomic/latest/reference/components/atomic-format-unit/